eZ/Publish/Core/Limitation/LocationLimitationType.php

```diff
@@ -125,25 +128,33 @@ public function evaluate(APILimitationValue $value, APIUserReference $currentUse
         } elseif ($object instanceof VersionInfo) {
             $object = $object->getContentInfo();
         } elseif (!$object instanceof ContentInfo) {
-            throw new InvalidArgumentException(
-                '$object',
-                'Must be of type: ContentCreateStruct, Content, VersionInfo or ContentInfo'
-            );
+            // As this is Role limitation we need to signal abstain on unsupported $object
```

Comment:
Only SectionLimitation and SubtreeLimitation are available when you assign a Role to the User or User Group. I am not saying that we should not return `ACCESS_ABSTAIN` but not for this reason. Also, this change is not needed to solve this issue. I am in favor of not making this change now.